### PR TITLE
Put modules into subfolders

### PR DESCRIPTION
**Pull request by MatBarba**
Also move subworkflows in subfolders and rename them to main

New structure:
```
├── modules
│   ├── database
│   │   ├── db_factory.nf
│   │   └── dump_db.nf
│   ├── download
│   │   ├── download_asm_data.nf
│   │   └── download_genbank.nf
│   ├── events
│   │   └── dump_events.nf
│   ├── fasta
│   │   └── process_fasta_data.nf
│   ├── files
│   │   ├── collect_files.nf
│   │   ├── collect_files_db.nf
│   │   └── publish_output.nf
│   ├── genbank
│   │   └── extract_from_gb.nf
│   ├── genome_metadata
│   │   ├── amend_genome_data.nf
│   │   └── prepare_genome_metadata.nf
│   ├── gff3
│   │   ├── gff3_validation.nf
│   │   ├── process_gff3.nf
│   │   └── unpack_gff3.nf
│   ├── manifest
│   │   ├── check_integrity_db.nf
│   │   ├── integrity.nf
│   │   ├── manifest_maker.nf
│   │   └── manifest_stats.nf
│   ├── schema
│   │   ├── check_json_schema.nf
│   │   ├── check_json_schema_db.nf
│   │   └── json_schema_factory.nf
│   ├── seq_region
│   │   ├── dump_seq_regions.nf
│   │   └── process_seq_region.nf
│   └── utils
│       └── utils.nf
├── subworkflows
│   ├── additional_seq_prepare
│   │   └── main.nf
│   ├── dump_metadata
│   │   └── main.nf
│   ├── dump_sql
│   │   └── main.nf
│   └── genome_prepare
│       └── main.nf
└── workflows
    ├── additional_seq_prepare
    │   ├── main.nf
    │   └── nextflow.config
    ├── dumper_pipeline
    │   ├── main.nf
    │   └── nextflow.config
    ├── genome_prepare
    │   ├── main.nf
    │   └── nextflow.config
    └── nextflow.config
```

### CODE REVIEW

File: pipelines/nextflow/modules/gff3/gff3_validation.nf [hunk 1]
```diff
@@ -30,8 +30,8 @@ process GFF3_VALIDATION {
 
   script:
   """
-  cp ${gene_models} gene_models.gff3.tmp 
-  gt gff3 -tidy -sort -retainids -force -o ${gene_models} gene_models.gff3.tmp 
-  gt gff3validator ${gene_models}
+  #cp ${gene_models} gene_models.gff3.tmp
```
[the remark this answers is not in the document]
Comment (MatBarba) [reply pipelines/nextflow/modules/gff3/gff3_validation.nf]: Nice spot! That wasn't supposed to make it to the commit :)

**Comment by ens-LCampbell** [comment]
From me, good to go ahead on PR merge thanks Matthieu :)

**Review by Dishalodha — APPROVED** [review]

**Comment by JAlvarezJarreta** [comment]
I would like to hear the rationale behind creating folders for subworkflows because having a folder for a single `main.nf` file does not look like the best approach to me (the rest looks great though).

**Comment by MatBarba** [comment]
> 

This is to imitate nf-core: the same folder is also meant to contain a file meta.yaml that contains the user documentation on how to use the subworkflow. Example: https://github.com/nf-core/rnaseq/tree/master/subworkflows/nf-core/fastq_align_hisat2

**Pull request merged**